docs/delivery_attempts.rst

```diff
@@ -311,6 +320,129 @@ and as the newer attempts complete, they will replace them.
    >>> attempt_50 in all_attempts
    True
 
+Automatic Deactivation on Failures
+----------------------------------
+
+If the history is completely filled up with failures, whether from
+validation errors, HTTP errors, or local processing errors, the
+subscription will be automatically marked inactive and future
+deliveries will not be attempted until it is manually activated again.
+
```

Comment:
My webhook endpoint goes down during the middle of the day, we drop a bunch of events on the floor (it looks like those basically get dropped if the subscription is inactive). I'm informed by an email when we deactivate the subscription and I fix the problem, reactivate the subscription. From that point I'm left to figure out how to deal with the dropped data? There is no record of it anywhere? 

We could conceivably "replay" hooks which we have a record of failure for, although ordering could get tricky. not sure what options we have here, but it sounds like users could be left with a big mess.

Reply:
> those basically get dropped if the subscription is inactive

Yes. I think they have to, otherwise we have designed a big DOS opportunity: subscribe to something popular, then start throwing 404 errors. Watch while the sending service fills up its disks and dies. 

>  not sure what options we have here, but it sounds like users could be left with a big mess.

It will depend on what the hook was for and what it does. Stuff like RTD just has a button to kick off whatever an incoming hook was supposed to do anyway. That works for anything that can derive what it needs from the current state (e.g., current master in that case). Other, more complex integrations, may have to fall back on a list query.

But basically, I don't think it's something that can or should be solved here. I think it's up to the consumer.